vllm/core/scheduler.py

```diff
@@ -279,8 +279,10 @@ def __init__(
         version = "v1"
         if self.scheduler_config.use_v2_block_manager:
             version = "v2"
-        if self.scheduler_config.embedding_mode:
+        if self.scheduler_config.model_mode == ModelMode.EMBEDDING:
             version = "embedding"
+        if self.scheduler_config.model_mode == ModelMode.SIMPLE:
+            version = "simple"
```

Comment:
Perhaps we should let each mode override the `BlockSpaceManagerImpl` instead of handling each case inside `Scheduler`.